src/libs/antares/study/parallel-years.h

```diff
@@ -0,0 +1,168 @@
+/*
+** Copyright 2007-2022 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+#pragma once
+
+#include <vector>
+
+namespace Antares
+{
+
+using namespace Data;
+
+struct setOfParallelYears
+{
+    // Un lot d'année à exécuter en parallèle.
+    // En fonction d'une éventuelle play-list, certaines seront jouées et d'autres non.
+
+public:
+    std::vector<unsigned int> setsSizes;
```

Comment:
**setsSizes** is used internally only. Shouldn't it be private ?
Besides, I do not understand the purpose of this variable.
Do we have to add this variable ?

Reply:
setOfParallelYears is a struct (not a class) so it behaves as a "bag of data" and doesn't really have usage for private members.
However the point on the utility of setsSizes is good. It acts as a "computed value"/cache value to avoid recalculating the value every time we need it. We only ever use the size of the vector as meaningful value so I change the type from vector to a simple uint counter.